src/app/routes/enrolment/enrolment-list/enrolment-list-filter/enrolment-list-filter.component.ts

```diff
@@ -50,20 +51,30 @@ export class EnrolmentListFilterComponent implements OnInit, OnDestroy {
     this.enrolmentFilterListService.setStatus(value);
   }
 
-  public async setFilters(showRevokeFilters: boolean) {
-    this.statusButtons = showRevokeFilters
-      ? [
-          FilterStatus.All,
-          FilterStatus.NotRevoked,
-          FilterStatus.RevokedOffChainOnly,
-          FilterStatus.Revoked,
-        ]
-      : [
-          FilterStatus.All,
-          FilterStatus.Pending,
-          FilterStatus.Approved,
-          FilterStatus.Rejected,
-          FilterStatus.Revoked,
-        ];
+  public async setFilters(viewType: EnrolmentView) {
+    const filtersByView = {
+      [EnrolmentView.MINE]: [
+        FilterStatus.All,
+        FilterStatus.Pending,
+        FilterStatus.Approved,
+        FilterStatus.Rejected,
+        FilterStatus.Revoked,
+        FilterStatus.Expired,
```

Comment:
I would suggest to move setting those filters to component above. 
What's more Revokable list do not have any more filters, just cascading filter (3 inputs for setting org/app/role name).

Reply:
and then passing in the filters to the component you mean?